app/src/androidTest/java/com/github/se/project/ui/lesson/RequestedLessonsTest.kt

```diff
@@ -146,76 +183,33 @@ class LessonsRequestedScreenTest {
   @Test
   fun testDatePickerButtonFunctionality() {
     composeTestRule.setContent {
-      RequestedLessonsScreen(listProfilesViewModel, lessonViewModel, navigationActions)
+      RequestedLessonsScreen(
+          listProfilesViewModel, lessonViewModel, mockNetworkStatusViewModel, navigationActions)
     }
 
     // Click on the DatePicker to ensure it can be interacted with
     composeTestRule.onNodeWithTag("datePicker").performClick()
   }
 
-  @Test
-  fun testLessonsFilteredBySelectedDate() {
-    // Set the selected date to filter lessons
-    val filteredDate = "2024-10-10"
-
-    // Change the lessons to include only those that match the date
-    requestedLessonsFlow.value = mockLessons.filter { it.timeSlot.contains(filteredDate) }
-
-    composeTestRule.setContent {
-      RequestedLessonsScreen(listProfilesViewModel, lessonViewModel, navigationActions)
-    }
-    // Verify that only the filtered lesson items are displayed
-    composeTestRule.onNodeWithText("Physics Tutoring").assertIsDisplayed()
-    composeTestRule.onNodeWithText("Math Tutoring").assertIsNotDisplayed()
-  }
-
   @Test
   fun testBottomNavigationSelection() {
     composeTestRule.setContent {
-      RequestedLessonsScreen(listProfilesViewModel, lessonViewModel, navigationActions)
+      RequestedLessonsScreen(
+          listProfilesViewModel, lessonViewModel, mockNetworkStatusViewModel, navigationActions)
     }
 
     // Interact with Bottom Navigation and verify navigation action
-    composeTestRule.onNodeWithTag("Find a Student").performClick()
+    composeTestRule.onNodeWithTag("centerElement").performClick()
 
```

Comment:
Important changes, here: first with the addition of the networkstatusviewmodel and then the new testtag, well done!